src/main/scala/epic/logo/ArgmaxInferencer.scala

```diff
@@ -0,0 +1,7 @@
+package epic.logo
+
+trait ArgmaxInferencer[T, Y, W] {
+
+  def argmax(weights : Weights[W], instance : T) : (Y, FeatureVector[W], Double)
```

Comment:
Maybe it should be private then. I'd advocate against pushing APIs that one doesn't want (or are redundant). Some kinds of technical debt is ok if eventually dealt with. Pushing breaking API changes to handle technical debt is not optimal.